chronosphere/tfschema/trace_common.go

```diff
@@ -168,7 +165,13 @@ var TraceStringFilterSchema = &schema.Schema{
 			},
 			"value": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+			},
+			"in_values": {
```

Comment:
nit: it might be a little confusing to see `NOT_IN` and then `IN_VALUES`. should we call it `values` and use OneOf to ensure either value or values is set, and add some custom validation such that values is only set for the multi-value match type?

Reply:
I see we already use in_values in the public API -- if we think we should use "values" we should do it in both. otherwise, can keep this as-is

Reply:
I found `values` to be a little bit to close to the longstanding `value` (single-value) field name; I pondered this and landed on `in_values` to make the connection to the `IN/NOT_IN` match type a bit more explicit.

BTW: regarding "OneOf" I assume you're talking about at the terraform layer (e.g., not public API protos) – does this refer to `ExactlyOneOf` / `AtLeastOneOf`? On that note, why couldn't I have previously used `ExactlyOneOf` to validate usage of consolidated sibling fields in #2 ?

Reply:
Yep, I was thinking of `ExactlyOneOf` and `AtLeastOneOf`. They're optional to give the user errors on the client-side, though we are moving towards server-side validation that returns errors as part of dry-run validation. 

If the dry-run validation works correctly, then we probably don't need to encode the same validation in the TF schema.

Reply:
I'm going to triple check the server side validation and will probably at least try adding the exactlyoneof validation for mutual exclusion over `value` and `in_values`

Reply:
the plot thickens:

```
// Only absolute attribute paths, ones starting with top level attribute
	// names, are supported. Attribute paths cannot be accurately declared
	// for TypeList (if MaxItems is greater than 1), TypeMap, or TypeSet
	// attributes. To reference an attribute under a single configuration block
	// (TypeList with Elem of *Resource and MaxItems of 1), the syntax is
	// "parent_block_name.0.child_attribute_name".
```

```
│ Internal validation of the provider failed! This is always a bug
│ with the provider itself, and not a user issue. Please report
│ this bug:
│ 
│ 3 errors occurred:
│ 	* resource chronosphere_trace_tail_sampling_rules: ExactlyOneOf: value
│ references unknown attribute (value) at part (value)
│ 	* resource chronosphere_dataset: ExactlyOneOf: value references unknown
│ attribute (value) at part (value)
│ 	* resource chronosphere_trace_metrics_rule: ExactlyOneOf: value references
│ unknown attribute (value) at part (value)
```

I think I just have to remove this because this nested type is reused everywhere and enumerating the absolute paths in every transitive usage will be error-prone. We'll just rely on server side validation.